api/lib/src/queries.rs

```diff
@@ -220,6 +220,21 @@ impl QueryApi {
 		result.next_block_in = Some(compute_distance(index, current_relative_slot, validator_len));
 		Ok(result)
 	}
+
+	pub async fn check_witnesses(
+		&self,
+		_block_hash: Option<state_chain_runtime::Hash>,
```

Comment:
Might be a good idea to allow querying historical blocks... This can be useful for debugging things in the past. It should be a relatively simple addition in the runtime rpc. LMK, I can do it on Monday.